frontend/src/views/WellsByRun.vue

```diff
@@ -0,0 +1,88 @@
+<script setup>
+import { onMounted, watch, ref, provide } from "vue"
+import { ElMessage } from "element-plus"
+import { useRoute, useRouter } from "vue-router"
+
+import LangQc from "@/utils/langqc"
+import WellTable from "@/components/WellTable.vue"
+import QcView from "@/components/QcView.vue"
+import QcControls from "@/components/QcControls.vue"
+
+import { generateUrl, qcQueryChanged } from "@/utils/url"
+import { useWellStore } from "@/stores/focusWell";
+import { getUserName } from "@/utils/session.js"
+
+const route = useRoute()
+const router = useRouter()
+const props = defineProps(["runName"])
+const focusWell = useWellStore()
+const serviceClient = new LangQc()
+
+let wellCollection = ref(null)
+let user = ref(null);
+let appConfig = ref(null)
+
+getUserName((email) => { user.value = email }).then();
+
+watch(() => route.query, (after, before) => {
+    if (qcQueryChanged(before, after) && (after['qcLabel'] || after['qcRun'])) {
+      focusWell.loadWellDetail(after.qcRun, after.qcLabel)
+    }
+  },
+  { immediate: true }
+)
+
+onMounted(() => {
+  serviceClient.getClientConfig().then(
+    data => {
+      appConfig.value = data
+    }
+  )
+
+  serviceClient.getWellsForRunPromise(props.runName).then(
+    (data) => wellCollection.value = data.wells
+  ).catch(error => {
+    ElMessage({
+      message: error.message,
+      type: "warning",
+      duration: 5000
+    })
+  })
+})
+
+provide('config', appConfig); // Needed by the QC widget
+
+function updateUrlQuery(newParams) {
+  // Merges the current URL query with new parameters
+  // The calling code does not need to replicate all properties from the
+  // previous URL query.
+  // route.query is read-only. We must copy it before modifying
+  const newUrl = generateUrl(Object.assign({}, route.query), newParams, route.path)
+
+  if (newUrl) {
+    router.push(newUrl)
+  }
+}
+
+function wellSelected(well) {
+  updateUrlQuery({qcRun: well.qcRun, qcLabel: well.qcLabel})
+}
+
+</script>
+
+<template>
+  <h2>Wells for run {{ props.runName }}</h2>
+  <WellTable :wellCollection="wellCollection" @wellSelected="wellSelected"/>
+  <h2>Well QC View</h2>
+  <div class="qcview" v-if="focusWell.runWell !== null">
+    <div class="data">
+      <QcView :runWell="focusWell.runWell" />
```

Comment:
QcView has no data fetching capability. It could operate with no properties at all and just access the focusWell itself, but I avoided that for making the testing more complex. Currently focusWell is manipulated by the QcControls, set by the well table, and observed by the QcView and I would prefer to keep it that way unless it becomes limiting.